packages/core/src/components/DataTable/index.tsx

```diff
@@ -214,11 +216,13 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
     if (!instantEdit) {
       const { changeLog }: { changeLog: ChangeLog } = this.state;
       const { originalIndex } = row.rowData.metadata;
-      if (Object.prototype.hasOwnProperty.call(changeLog, originalIndex)) {
-        changeLog[originalIndex][key] = newVal;
+
+      if (changeLog[originalIndex] && changeLog[originalIndex][key]) {
+        changeLog[originalIndex][key] = { newVal };
       } else {
-        changeLog[originalIndex] = { [key]: newVal };
+        changeLog[originalIndex] = { [key]: { newVal } };
```

Comment:
The types had it as `{ newVal: string }` but the code was just `newVal`. Not sure which is correct.